cmake/ThrustBuildCompilerTargets.cmake

```diff
@@ -20,27 +31,45 @@ function(thrust_build_compiler_targets)
   if (THRUST_TBB_FOUND)
     # There's a ton of these in the TBB backend, even though the code is correct.
     # TODO: silence these warnings in code instead
-    append_option_if_available("-Wno-unused-parameter" cxx_compile_options)
+#    append_option_if_available("-Wno-unused-parameter" cxx_compile_options)
```

Comment:
Why comment this out? If we no longer need it, can we just remove the flag?

Reply:
That's the plan -- this PR is marked as a draft because it's still a work in progress.

Reply:
Looks like this is still commented out and needs to be re-added.